src/e2e-test/features/gcs/source/GCSSourceToBigQuery.feature

```diff
@@ -285,3 +285,304 @@ Feature: GCS source - Verification of GCS to BQ successful data transfer
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Enter runtime argument value "gcsJsonFileSchema" for key "OutSchema"
+    Then Run the preview of pipeline with runtime arguments
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "gcsJsonFileSchema" for key "OutSchema"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Get count of no of records transferred to target BigQuery Table
+
+  @GCS_CSV_TEST @BQ_SINK_TEST
+  Scenario: To verify Successful GCS to BigQuery data transfer with enable data file encryption flag true
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "GCS" from the plugins list as: "Source"
+    When Sink is BigQuery
+    Then Connect source as "GCS" and sink as "BigQuery" to establish connection
+    Then Open GCS source properties
+    Then Enter GCS property projectId and reference name
+    Then Override Service account details if set in environment variables
+    Then Enter GCS source property path "gcsCsvFile"
+    Then Select GCS property format "csv"
+    Then Toggle GCS source property skip header to true
+    Then Validate output schema with expectedSchema "gcsCsvFileSchema"
+    Then Validate "GCS" plugin properties
+    Then Select radio button plugin property: "encrypted" with value: "true"
+    Then Close the GCS properties
+    Then Open BigQuery sink properties
+    Then Override Service account details if set in environment variables
+    Then Enter the BigQuery sink mandatory properties
+    Then Validate "BigQuery" plugin properties
+    Then Close the BigQuery properties
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Get count of no of records transferred to target BigQuery Table
+
+#Macro Test Scenario
+  @CMEK @GCS_CSV_TEST @BQ_SINK_TEST @BigQuery_Sink_Required
+  Scenario:Validate successful records transfer from GCS to BigQuery with macro arguments at source_master
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "GCS" from the plugins list as: "Source"
+    When Sink is BigQuery
```

Comment:
Use the new step similar to above for selecting the Bigquery plugin from sink

Reply:
Done.